staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -138,6 +144,56 @@ func (w *watcher) createWatchChan(ctx context.Context, key string, rev int64, re
 	return wc
 }
 
+// getStartWatchResourceVersion returns a ResourceVersion
+// the watch will be started from.
+// Depending on the input parameters the semantics of the returned ResourceVersion are:
+//   - start at Exact (return resourceVersion)
+//   - start at Most Recent (return an RV from etcd)
+func (w *watcher) getStartWatchResourceVersion(ctx context.Context, resourceVersion int64, opts storage.ListOptions) (int64, error) {
+	if resourceVersion > 0 {
+		return resourceVersion, nil
+	}
+	if !utilfeature.DefaultFeatureGate.Enabled(features.WatchList) {
+		return resourceVersion, nil
+	}
+	if opts.SendInitialEvents == nil || *opts.SendInitialEvents {
+		return resourceVersion, nil
```

Comment:
hmm - looking deeper into it, I'm not sure I'm following this case:
<invariant> we know that RV=0 at this point
a) SendInitialEvents=nil => this is legacy case and we indeed want to start from rv=0
b) SendInitialEvents=true => for this case shouldn't we return the "currentStorageRV" ?

Reply:
For `SendInitialEvents=true` it is kind of implicit because we will issue a consistent list against etcd followed by the bookmark which will include the `currentStorageRV`

Reply:
OK - let's add a comment that this is handled by getting the state from etcd.

And for the below case, let's make an explicit comment that only watch stream will be handled there.